### PR TITLE
[core] add deletedFiles in NewFilesIncrement

### PR DESCRIPTION
**Pull request by YannByron**
### Purpose

Linked issue: close #xxx

### CODE REVIEW

File: paimon-core/src/main/java/org/apache/paimon/mergetree/MergeTreeWriter.java [hunk 1]
```diff
@@ -305,7 +311,9 @@ public void close() throws Exception {
 
         // delete temporary files
         List<DataFileMeta> delete = new ArrayList<>(newFiles);
+        delete.addAll(deletedFiles);
```
[the remark this answers is not in the document]
Comment (YannByron) [reply paimon-core/src/main/java/org/apache/paimon/mergetree/MergeTreeWriter.java:314]: OK, but we still need to clear `deletedFiles`, right ?

**Comment by JingsongLi** [comment]
+1

**Comment by YannByron** [comment]
https://github.com/apache/paimon/issues/3057

**Comment by nonggialiang** [comment]
@YannByron @JingsongLi 
Shall we increase the CommitMessageSerializer.CURRENT_VERSION and deserialize the objects according to the version for backward compactibility support ?

**Pull request merged**